yudiencore/core.go

```diff
@@ -233,22 +248,24 @@ func UdnDebugReset(udn_schema map[string]interface{}) {
 }
 
 func UdnDebugIncrementChunk(udn_schema map[string]interface{}) {
-	current := udn_schema["debug_html_chunk_count"].(int)
-	current++
-	udn_schema["debug_html_chunk_count"] = current
+	if (Debug_Udn || udn_schema["udn_debug"].(bool)) {
```

Comment:
This function should be called with the assumption that the debug is already on. The call to the function itself should be wrapped with a debug flag check. 

Otherwise, the UdnDebug* like this will get called and not have outward behavior, so it will be a bit harder to understand its behavior while reading. 

Also -- `udn_schema["udn_debug"]` -- the purpose is a bit unclear to me, can you explain?

Reply:
udn_schema["udn_debug"] is the same as Debug_Udn. Say in the code Debug_Udn is turned off but during the execution you want to turn debug on, you could used UDN to turn debug on by toggling udn_schema without any change to the code